From 8cacfc85b615cc0bae01241593c4b25da6570efc Mon Sep 17 00:00:00 2001
Message-Id: <8cacfc85b615cc0bae01241593c4b25da6570efc.1648733020.git.geliang.tang@suse.com>
From: Colin Ian King <colin.king@canonical.com>
Date: Tue, 17 Aug 2021 18:08:42 +0100
Subject: [PATCH] bpf: Remove redundant initialization of variable allow
Git-commit: 8cacfc85b615cc0bae01241593c4b25da6570efc
Patch-mainline: v5.15-rc1
References: bsn#127
Modified-by-SEL: No


The variable allow is being initialized with a value that is never read, it
is being updated later on. The assignment is redundant and can be removed.

Addresses-Coverity: ("Unused value")

Signed-off-by: Colin Ian King <colin.king@canonical.com>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Link: https://lore.kernel.org/bpf/20210817170842.495440-1-colin.king@canonical.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 kernel/bpf/cgroup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/bpf/cgroup.c b/kernel/bpf/cgroup.c
index a1dedba4c174..9f35928bab0a 100644
--- a/kernel/bpf/cgroup.c
+++ b/kernel/bpf/cgroup.c
@@ -1135,7 +1135,7 @@ int __cgroup_bpf_check_dev_permission(short dev_type, u32 major, u32 minor,
 		.major = major,
 		.minor = minor,
 	};
-	int allow = 1;
+	int allow;
 
 	rcu_read_lock();
 	cgrp = task_dfl_cgroup(current);
-- 
2.34.1

